core-library/src/main/java/org/silverpeas/core/contribution/contentcontainer/content/AbstractContentInterface.java

```diff
@@ -0,0 +1,473 @@
+
+  protected ContentManager getContentManager() {
+    return ContentManagerProvider.getContentManager();
+  }
+
+  @Override
+  public final List<SilverContentInterface> getSilverContentById(
+      final List<Integer> silverpeasContentIds, final String instanceId,
+      final String currentUserId) {
+    List<String> resourceIds =
+        getContentManager().getResourcesMatchingContents(silverpeasContentIds);
+    List<? extends Contribution> contributions =
+        getAccessibleContributions(resourceIds, instanceId, currentUserId);
+    return contributions.stream().map(instance -> convert(instance, instanceId))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public <T extends Contribution> int getSilverContentId(final T contribution) {
+    final ContributionIdentifier contributionId = contribution.getContributionId();
+    return getSilverContentId(contributionId.getLocalId(), contributionId.getComponentInstanceId());
+  }
+
+  @Override
+  public <T extends Contribution> int getOrCreateSilverContentId(final T contribution) {
+    int contentId = getSilverContentId(contribution);
+    if (contentId == -1) {
+      contentId = createSilverContent(contribution, contribution.getLastModifier().getId());
+    }
+    return contentId;
+  }
+
+  @Override
+  public int getOrCreateSilverContentId(final String resourceId, final String componentInstanceId) {
+    Contribution contribution = getContribution(resourceId, componentInstanceId).orElseThrow(
+        () -> new IllegalArgumentException(MessageFormat
+            .format("impossible to get contribution from id {0} into component instance {1}",
+                resourceId, componentInstanceId)));
+    return getOrCreateSilverContentId(contribution);
+  }
+
+  /**
+   * <p>
+   * Gets the contribution from its identifier and the identifier of the component instance.
+   * </p>
+   * <p>
+   * The user rights are not verified here as this method must be used only for silverpeas content
+   * id management and not to provide the data to an other service.
+   * </p>
+   * @param resourceId a resource identifier.
+   * @param componentInstanceId a component instance identifier.
+   * @return an optional {@link Contribution} instance.
+   */
+  protected abstract Optional<Contribution> getContribution(final String resourceId,
+      final String componentInstanceId);
+
+  /**
+   * <p>
+   * Gets list of contribution from their identifier and the identifier of the component instance.
+   * </p>
+   * <p>
+   * Implementations of this method must handle the user rights on the resource.
+   * </p>
+   * @param resourceIds a list of resource identifier.
+   * @param componentInstanceId a component instance identifier.
+   * @param currentUserId the identifier of the user accessing the content.
+   * @return an optional {@link Contribution} instance.
+   */
+  protected abstract List<? extends Contribution> getAccessibleContributions(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1452)